roster2/tool/src/java/org/sakaiproject/roster/impl/SakaiProxyImpl.java

```diff
@@ -297,10 +297,15 @@ public Boolean getViewEmail(String siteId) {
     }
 
     @Override
-    public Boolean  getViewUserDisplayId() {
+    public Boolean getViewUserDisplayId() {
+        return getViewUserDisplayId(getCurrentSiteId());
+    }
+
+    @Override
+    public Boolean  getViewUserDisplayId(String siteId) {
         return serverConfigurationService.getBoolean(
                 "roster.display.userDisplayId", DEFAULT_VIEW_USER_DISPLAY_ID) &&
-                hasUserSitePermission(getCurrentUserId(), RosterFunctions.ROSTER_FUNCTION_VIEWID, getCurrentSiteId());
+                hasUserSitePermission(getCurrentUserId(), RosterFunctions.ROSTER_FUNCTION_VIEWID, siteId);
```

Comment:
This may still need to NPE check on siteId and just return what was previouly there if it's null? 

```
if (siteId == null) 
	return serverConfigurationService.getBoolean("roster.display.userDisplayId", DEFAULT_VIEW_USER_DISPLAY_ID);
```

Reply:
well it's the getCurrentSiteId() that will throw a big error